src/Client.php

```diff
@@ -25,10 +25,13 @@ class Client extends ElasticaClient
     private $indexer;
     private $indexBuilder;
 
+    /** @var AbstractConfiguration $configuration */
+    private $configuration;
+
     public function getIndexBuilder(): IndexBuilder
     {
         if (!$this->indexBuilder) {
-            $this->indexBuilder = new IndexBuilder($this, $this->getConfig(self::CONFIG_MAPPINGS_DIRECTORY));
+            $this->indexBuilder = new IndexBuilder($this, $this->getConfiguration());
```

Comment:
Maybe having "mapping reader" is better for understanding the code here?

```suggestion
            $this->indexBuilder = new IndexBuilder($this, $this->getMappingReader());
```

Reply:
fixed